src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp

```diff
@@ -615,39 +609,51 @@ void AlgorithmPaintToolBox::updateView()
     if (view)
     {
         setCurrentView(qobject_cast<medAbstractImageView*>(view));
-    }
-    updateMouseInteraction();
 
-    // TODO : get rid of similar lines in mousePressEvent, we need this here otherwise we cannot use interpolate/copy/paste
-    //                     //  on the view as long as the setData is not called for this view
-    if (currentView)
-    {
+        updateMouseInteraction();
+
         medAbstractData* data = currentView->layerData(0);
-        if(!data)
-            return;
-        medImageMaskAnnotationData * existingMaskAnnData = dynamic_cast<medImageMaskAnnotationData *>(data);
-        if(!existingMaskAnnData)
+
+        if(data)
         {
-            setData( data );
-        }
+            medImageMaskAnnotationData * existingMaskAnnData = dynamic_cast<medImageMaskAnnotationData *>(data);
+            if(!existingMaskAnnData)
+            {
+                setData( data );
+            }
 
-        // Update cursor if the orientation change
-        connect(currentView, SIGNAL(orientationChanged()), this, SLOT(activateCustomedCursor()), Qt::UniqueConnection);
+            // Update cursor if the orientation change
+            connect(currentView, SIGNAL(orientationChanged()), this, SLOT(activateCustomedCursor()), Qt::UniqueConnection);
 
-        connect(currentView,SIGNAL(layerRemoved(unsigned int)),this,SLOT(closeView()), Qt::UniqueConnection);
-    }
+            connect(currentView,SIGNAL(layerRemoved(unsigned int)),this,SLOT(clear()), Qt::UniqueConnection);
 
-    // Update cursor to new view
-    if ( this->m_strokeButton->isChecked() )
-    {
-        activateCustomedCursor(); // Add circular cursor for painting
+            // Update cursor to new view
+            if ( this->m_strokeButton->isChecked() )
+            {
+                activateCustomedCursor(); // Add circular cursor for painting
+            }
+            else
+            {
+                deactivateCustomedCursor(); // Deactivate painting cursor
+            }
+
+            setDisabled(false);
+        }
     }
     else
     {
-        deactivateCustomedCursor(); // Deactivate painting cursor
+        setDisabled(true);
+        currentView = NULL;
     }
 }
 
+void AlgorithmPaintToolBox::setDisabled(bool disable)
```

Comment:
> void QWidget::setDisabled(bool disable)
> Disables widget input events if disable is true; otherwise enables input events.
Hum, indeed. @fcollot are you ok if we go back to `disableButtons(bool)` ? XD